internal/shared/kubernetes/kubernetes.go

```diff
@@ -75,3 +83,33 @@ func GetSecret(ctx context.Context, client k8sClient.Client, namespace, secretNa
 	}
 	return secret, nil
 }
+
+// ApplyOwnershipReference adds  or updates ownerships to kubernetes resources in GitSync CRD
+func ApplyOwnershipReference(manifest string, gitSync *v1alpha1.GitSync) ([]byte, error) {
+	decode := scheme.Codecs.UniversalDeserializer().Decode
+	decoded, _, err := decode([]byte(manifest), nil, nil)
+	if err != nil {
+		return nil, err
+	}
+	obj, ok := decoded.(metav1.Object)
+	if !ok {
+		return nil, errors.New("decoded manifest is not a metaV1 object")
+	}
+	ownerRef := metav1.OwnerReference{
+		APIVersion:         gitSync.APIVersion,
+		Kind:               gitSync.Kind,
+		Name:               gitSync.Name,
+		UID:                gitSync.UID,
+		Controller:         ptr.To(true),
+		BlockOwnerDeletion: ptr.To(true),
+	}
+	existingRef := obj.GetOwnerReferences()
+	existingRef = append(existingRef, ownerRef)
+	obj.SetOwnerReferences(existingRef)
```

Comment:
I want to make sure we don't possibly add the same OwnerReference when it's already in there. If there's any possibility, then we should check before we add it.

Reply:
Also a test case for that